src/rgw/cls_fifo_legacy.cc

```diff
@@ -784,6 +784,7 @@ int FIFO::_prepare_new_part(const DoutPrefixProvider *dpp,
     r = _update_meta(dpp, u, version, &canceled, tid, y);
     if (r >= 0 && canceled) {
       std::unique_lock l(m);
+      version = info.version;
```

Comment:
Instead of sprinkling version renewal here and there, why don't we unify all ops (create_part, trim, ...) get the version while creating updata_meta op?

I was working on your Nov 1 head (https://github.com/ceph/ceph/commit/cfc3bde36dbc9c6e0b7182bbb32539019ba2dd61) and making some changes on top. Below is the relevant commit as an example:

https://github.com/adamemerson/ceph/pull/3/commits/0dd5676e73368ef43391f5100313ffd10e02ea13

This way noone misses to update the version before trying to update_meta and then apply_update.

Reply:
The rationale is that we want the version we have to correspond with the other data we fetched while under lock that we're acting on.